src/python/WMComponent/AgentStatusWatcher/DrainStatusPoller.py

```diff
@@ -123,11 +140,13 @@ def checkSpeedDrainThresholds(self):
         # loop through the speed drain configuration and make a list of what thresholds have been hit
         for k, v in self.speedDrainConfig.items():
             # make sure keys in the speed drain config are valid
-            if k in validKeys and isinstance(v['Threshold'], int) and isinstance(v['Enabled'], bool):
+            if k in self.validSpeedDrainConfigKeys and isinstance(v['Threshold'], int) and isinstance(v['Enabled'], bool):
                 # we always want to apply the condor priority change if the threshold is hit
                 if not v['Enabled'] or k == 'CondorPriority':
                     logging.info("Checking speed drain threshold for %s. ", k)
-                    if len(jobs) < v['Threshold']:
+                    numJobs = len(jobs)
+                    # Don't change if  there is no jobs
+                    if numJobs > 0 and numJobs < v['Threshold']:
```

Comment:
Sorry Seangchan, I might have missed another use case yesterday. EnableAllSites should be evaluated even when there are no jobs in the condor queue. Reason is, there might be jobs stuck in the JobSubmitter cache waiting for that site to come back. Even though I'm not completely sure that can happen nowadays...

Reply:
I don't think we have to worry that much, but since if this happen it can get stuck long time. I will remove that check. We have other protection anyway. Good catch, Alan.